ci-build.sh

```diff
@@ -67,6 +67,35 @@ SRC_DIR=$(pwd)
 mkdir -p "build-${CC}-${PROTOCOL}"
 cd "build-${CC}-${PROTOCOL}"
 
+# Check to see if we _just_ tested this rev in
+# a merge queue, and if so don't bother doing
+# it again. Wastes billable CPU time.
+if [ -e prev-pass-rev ]
+then
+   PREV_REV=$(cat prev-pass-rev)
+   CURR_REV=$(git -C "${SRC_DIR}" rev-parse HEAD)
+   if [ "${PREV_REV}" = "${CURR_REV}" ]
+   then
+       exit 0
+   fi
+   rm -f prev-pass-rev
+fi
+
+
+# restore source file mtimes based on content hashes
```

Comment:
FYI we do the same time traveling to improve reproducibility for Go builds here: https://github.com/stellar/go/blob/f7134ebcdbfeff3f38d9f0506d1100037a44d921/.github/actions/setup-go/action.yml#L28

The approach is a bit different. Not sure if that's helpful, I don't know if there's benefits to either approach. 

This looks good though.

Reply:
ah, git-restore-mtime might be better. I will give that a try in a followup.